File: app/src/main/java/com/example/sony/newsapi/repository/RepositoryImpl.java
package com.example.sony.newsapi.repository;

import android.support.annotation.NonNull;

import com.example.sony.newsapi.model.Article;

import java.util.List;

import io.realm.Realm;

public class RepositoryImpl implements Repository {
    @Override
    public void saveToDB(final List<Article> articles) {
        Realm realm = Realm.getDefaultInstance();
        realm.executeTransaction(new Realm.Transaction() {
            @Override
            public void execute(@NonNull Realm realm) {
                realm.copyToRealmOrUpdate(articles);
            }
        });
        realm.close();
    }

    @Override
    public List<Article> loadFromDB() {
        Realm realm = Realm.getDefaultInstance();
        List<Article> articles = realm.copyFromRealm(realm.where(Article.class).findAll());
        realm.close();
        return articles;
    }

    @Override
    public List<Article> loadFromApi() {
        return null;
    }
}

File: app/src/main/java/com/example/sony/newsapi/repository/RepositoryImpl.kt
package com.example.sony.newsapi.repository

import com.example.sony.newsapi.model.Article

import io.realm.Realm

class RepositoryImpl : Repository {
    override fun saveToDB(articles: List<Article>) {
        val realm = Realm.getDefaultInstance()
        realm.executeTransaction { realm1 -> realm1.copyToRealmOrUpdate(articles) }
        realm.close()
    }

    override fun loadFromDB(): List<Article> {
        val realm = Realm.getDefaultInstance()
        val articles = realm.copyFromRealm(realm.where(Article::class.java).findAll())
        realm.close()
        return articles
    }

    override fun loadFromApi(): List<Article> {
        TODO("not implemented")
    }
}
